smartsim/_core/launcher/dragon/dragonBackend.py

```diff
@@ -620,7 +607,7 @@ def _refresh_statuses(self) -> None:
                         except KeyError:
                             logger.error(f"Tried to free a non-allocated host: {host}")
                         self._free_hosts.append(host)
-                    group_info.process_group = None
+                    # group_info.process_group = None
                     group_info.redir_workers = None
```

Comment:
Question: Is there a reason that we were previously nulling out this attr? I needed to do this to make sure that return codes did not come back as `[-1]`, but if there is a reason for doing this I can makes something a bit more sophisticated for how I retrieve return codes (e.g. cache them and then null out the process group).

Reply:
Uhh... I know that it works _behaviorally_ but I've not checked for actual resource utilization (e.g. does this need to be nulled out in order for dragon to garbage collect, etc.). I might need to take some time to take a look at that!

Reply:
Changed to cache return codes when marking process complete!